File: dataset/human/cpp/file_1041.cpp
#include<iostream>
using namespace std;
int main () {
  int R, G, B, N;
  cin >> R >> G >> B >> N;
  int ans = 0;
  for (int i = 0; i <= N; i ++) {
    for (int j = 0; j <= N; j ++) {
      int A = R * i + G * j;
      int n = N - A;
      //cout << n << endl;
      if (n >= 0 && n % B == 0) ans ++;
    }
  }
  cout << ans << endl;
}

File: dataset/human/python/file_1041.py
import sys
def input():
    return sys.stdin.readline()[:-1]
from collections import *
from math import atan2, pi
from bisect import *
mod = 10**9+7

N = int(input())
S = [int(c == 'B') for c in input()]
if S[0] + S[-1] < 2:
    print(0)
    exit()
for i in range(N):
    S[2*i] ^= 1
if sum(S) != N:
    print(0)
    exit()
cnt, ans = 0, 1
for x in S:
    if x == 0:
        cnt += 1
    else:
        if cnt == 0:
            print(0)
            exit()
        ans = ans * cnt % mod
        cnt -= 1
for i in range(2, N+1):
    ans = ans * i % mod
print(ans)
